haskell/test/Test/TimedOracles.hs

```diff
@@ -0,0 +1,356 @@
+-- SPDX-FileCopyrightText: 2021 Arthur Breitman
+-- SPDX-License-Identifier: LicenseRef-MIT-Arthur-Breitman
+
+module Test.TimedOracles
+  ( module Test.TimedOracles
+  ) where
+
+import Prelude
+
+import qualified Data.List as List
+import Fmt (pretty, (+|), (|+))
+import Lorentz.Macro hiding (assert)
+import Lorentz.Test (contractConsumer, sec)
+import Lorentz.Value
+import Morley.Nettest
+import Morley.Nettest.Tasty
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.HUnit (testCase, (@?=))
+import Tezos.Core (timestampPlusSeconds)
+
+import SegCFMM.Errors
+import SegCFMM.Types
+import Test.Invariants
+import Test.SegCFMM.Storage (defaultStorage)
+import Test.Util
+
+test_BufferInitialization :: TestTree
+test_BufferInitialization =
+  testCase "Our initial buffer matches the ligo's one for buffer size = 1" $
+    pretty @_ @Text (initCumulativesBuffer 0) @?= pretty (sCumulativesBuffer defaultStorage)
+
+test_Continuity :: TestTree
+test_Continuity =
+  nettestScenarioCaps "Returned cumulative values continuously grow over time" do
+    alice <- newAddress "alice"
+    (cfmm, _) <- prepareSomeSegCFMM defaultStorage
+      { sCumulativesBuffer = initCumulativesBuffer 100
+      } alice
+
+    advanceLevel 3
+
+    withSender alice do
+      call cfmm (Call @"Set_position") $
+        setPositionParamSimple (TickIndex (-100), TickIndex 100) 100
+
+    advanceLevel 3
+
+    checkedTimes <- do
+      cumulatives <- getStorage cfmm >>= lastRecordedCumulatives
+      let time = tcTime cumulatives
+      return $ timestampPlusSeconds time <$> [-3 .. 0]
+
+    -- Our property of interest here:
+    -- lim{t -> record_time} cumulative(t) = cumulative(record_time)
+    -- We will also check places of regular growth for the company.
```

Comment:
WDYM by "for the company" here?

Reply:
I meant "along with that", the current wording is really confusing :unamused: